### PR TITLE
Configure V2 to be case insensitive

### PR DESCRIPTION
**Pull request by gortiz**
While V1 is case insensitive, V2 is case sensitive.

I'm not sure if we consider this a bug or a feature. This PR assumes it is a bug and therefore changes the V2 behavior to be case insensitive.

I've also modified `OfflineClusterIntegrationTest` to run the case insensitivity tests with V2. Given that these tests were testing the case insensitivity but also the fact that V1 ignores the schema, I had to split them:
* testCaseInsensitivity and testCaseInsensitivityWithColumnNameContainsTableName test case sensibility
* testSchemaIgnoredInV1 and testIgnoresSchemaInColumnNameWithTestNameInV1 test that schema is ignored in V1

In case we consider case sensibility in V2 a feature, feel free to close this PR

### CODE REVIEW

**Comment by gortiz** [comment]
@xiangfu0 @Jackie-Jiang

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java [hunk 1]
```diff
@@ -648,7 +671,7 @@ public Schema build() {
       } catch (Exception e) {
         throw new RuntimeException("Invalid schema", e);
       }
-      return _schema;
+      return new Schema(_schema);
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java:674]: Is this related? Why do we want to do a swallow copy?

Comment (gortiz) [reply pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java:674]: This is not completely related, but it was incorrect before. Specifically, there are several tests that do something like:

```java
    TABLE_SCHEMAS.put("a_REALTIME", SCHEMA_BUILDER.setSchemaName("a").build());
    TABLE_SCHEMAS.put("b_REALTIME", SCHEMA_BUILDER.setSchemaName("b").build());
    TABLE_SCHEMAS.put("c_OFFLINE", SCHEMA_BUILDER.setSchemaName("c").build());
    TABLE_SCHEMAS.put("d", SCHEMA_BUILDER.setSchemaName("d").build());
```

The above code is from `QueryEnvironmentTestBase`, but there are others that do similar things.

I've discovered that after debugging for a while trying to understand why table `A` had schema `D` :facepalm: 

We can either apply the shallow copy in another PR or change the affected tests
